projects/packages/my-jetpack/_inc/components/welcome-banner/index.jsx

```diff
@@ -17,10 +17,10 @@ import styles from './style.module.scss';
 const WelcomeBanner = () => {
 	const isNewUser = window.myJetpackInitialState.userIsNewToJetpack === '1';
 	const { recordEvent } = useAnalytics();
-	const { hasBeenDismissed, dismissWelcomeBanner } = useWelcomeBanner();
+	const { isDismissed, dismissWelcomeBanner } = useWelcomeBanner();
 	const { isRegistered, isUserConnected } = useConnection();
 	const navigateToConnectionPage = useMyJetpackNavigate( '/connection' );
-	const [ bannerVisible, setBannerVisible ] = useState( ! hasBeenDismissed && isNewUser );
+	const [ bannerVisible, setBannerVisible ] = useState( ! isDismissed && isNewUser );
```

Comment:
AFAIK this sets the initial state, and will not get updated when the variable `isDismissed` is updated. That might be the issue with the original code. We seem to have never been using the result of the fetch query (only initial window state).

One solution is to add:
```
useEffect( () => { setBannerVisible( ! isDismissed && isNewUser ) }, [ isDismissed, isNewUser ] );
```

Reply:
I got confused about the code, all works correctly!